### ISSUE TITLE
[Navigator] Push transition, NavigationBar's buttons can be clicked and triggered more than one time

### ISSUE THREAD

**Issue by everfire130** [facebook/react-native#2048]
for example,if my NavigationBar's right button's function is push to next route,I click the button twice quickly,it will push twice

**Comment by cyprusglobe** [comment]
I'm seeing this as well

**Comment by chirag04** [comment]
@ericvicenti I did see this happen in one of my app as well. The bug here is that at the end you would see a white screen.

**Comment by brentvatne** [comment]
I've assigned @ericvicenti but he is a busy fellow so if anybody would like to take a crack at fixing this it would be very very welcome! @everfire130 @cyprusglobe @chirag04

**Comment by ide** [comment]
Anyone have ideas for how to solve this? Best one I have so far is to block touch events on the nav bar elements if they're being transitioned out.

**Comment by ericvicenti** [comment]
Here are a few different ways to solve this, in order of my preference:

1. The tap events could be made idempotent- "push this route if it is not already present"
2. Navigator could tell the navbar about transition state and the navbar would ignore touches during transitions
3. Navigator could block all touches to the navbar during transitions
4. Navigator could warn or silently fail when attempting to push during a transition

Further discussion is welcome.

**Comment by ide** [comment]
I believe 1 wouldn't work for a lot of apps like instagram where there isn't a canonical route instance for each kind of scene. For example in instagram you can navigate from feed -> photo -> profile -> same photo, and the nav stack needs to have two different route objects for the same photo scene. To implement each transition, I would create a new route object and push it onto the nav stack.

So when the user is on the feed screen and taps the same photo twice, I would create two separate route objects (both corresponding to a screen for the same photo) and push them both onto the stack, and Navigator would be unable to tell that the routes should be deduped in some way.

I believe Apple's solution is to ignore touches on inactive scenes and inactive navbar content. I'm not sure if they block all user interaction with the nav bar or just the nav buttons that are transitioning away, while preserving user interaction with the new buttons transitioning in.

**Comment by everfire130** [comment]
@brentvatne @ide @ericvicenti  Thanks for your comments  and suggestions.I believe you would solved this problem soon in the next version.Support you!O(∩_∩)O~~Now my temporary solution is to toggle the navigation bar button function through navigator life cycle .My English is weak.I hope it can be understand.😁

**Comment by ericvicenti** [comment]
@ide, to do something like 1 with the Instagram example, we could do something like this:

```
getInitialState: function() {
  this.photoRoute = new IGPhotoRoute({ photo: this.props.photo.id });
  return {};
},
handlePush: function() {
  this.props.navigator.push(this.photoRoute);
  // the navigator should navigate to the the photoRoute if it is already in the stack
},
```

**Comment by satya164** [comment]
Actually, it seems that if the navbar had a button on the right, and I pushed a route without a right button, I can still click on the right and it works. Weird!

Any progress on this? may be we could block touch events when the scene is not in focus?

**Comment by bobbobb2118** [comment]
I need assistance in getting this Facebook taken down and kept down
Innappropriate content meant to harrass

**Comment by mkonicek** [comment]
Hi there! This issue is being closed because it has been inactive for a while.

But don't worry, it will live on with ProductPains! Check out its new home: https://productpains.com/post/react-native/navigator-push-transition-navigationbars-buttons-can-be-clicked-and-triggered-more-than-one-time

ProductPains helps the community prioritize the most important issues thanks to its voting feature.
It is easy to use - just login with GitHub.

Also, if this issue is a bug, please consider sending a PR with a fix. We rely on the community to provide
bugfixes as the core team is focused on performance.

**Comment by mfuruskog** [comment]
It seems like this has not been fixed as I have this problem with 0.23. Did someone come up with a fix for this?

**Comment by ashleydw** [comment]
@mfuruskog I get this too on 0.24.1, the simple solution is to always return a view in `RightButton` / `LeftButton` (be sure to set width/height properties or it will render as 0x0)
